src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
+pub struct State1 {
+    channel: Channel,
+    final_address_self: Address,
+    final_address_other: Address,
+}
+pub struct FinalState(Transaction);
```

Comment:
:upside_down_face: I would just name it `State2`.

At a higher level we can ensure that we broadcast this transaction and either return nothing at all or a `ClosedChannel` final struct.

The name is just cosmetic though, so it doesn't matter too much.

Reply:
I'd even go further and say that the last action should only return a `Transaction` without a need to newtype it.

Reply:
> I'd even go further and say that the last action should only return a `Transaction` without a need to newtype it.

I'm not 100% sure yet. See comment in the PR description: 
> I introduce a final state close::FinalState from which one should not be able to update the channel anymore as it is not possible to invalidate a signed close transaction anymore.
> For safety reasons it might be necessary to keep the revoked states as well but I left this out for now.

Depending on how the library evolves we might need to keep track of more information such as the revocation keys.

Reply:
It's best not to chose a design A in case we might want later to do some change that would justify A.

Only return the `Transaction`, if we decide later to also return the revoked states then we can change the API of this function to return a new struct `FinalState` that contains the transaction and the revoked states.

Similarly to an argument I am making today, in this case, I would still not use a `FinalState` struct but a tuple as I would expect the consumer of this function to deconstruct the returned variable to broadcast the transaction upon execution of the function.

Reply:
> It's best not to chose a design A in case we might want later to do some change that would justify A.

I disagree with this statement :) 
Imho if it's not clear how the Api will evolve go with whatever is the fastest. At the time of writing everything is a bet.